### PR TITLE
Add reverse' test

### PR DESCRIPTION
**Pull request by harendra-kumar**
Add more diagnostic information to debug better when tests fail.

Print the lists being compared in the fromListN test.

Added list length printing to know the length when the lists printed are large.

### CODE REVIEW

File: test/lib/Streamly/Test/Prelude/Common.hs [hunk 1]
```diff
@@ -1180,7 +1181,7 @@ transformCombineOpsCommon constr desc eq t = do
 
     -- reordering
     prop (desc <> " reverse") $ transform reverse t S.reverse
-    -- prop (desc <> " reverse'") $ transform reverse t S.reverse'
+    prop (desc <> " reverse'") $ transform reverse t S.reverse'
```
Comment (pranaysashank) [inline test/lib/Streamly/Test/Prelude/Common.hs:1184]: This only works on ordered streams, so I don't think it should be in `transformCombineOpsCommon`

**Review by pranaysashank — CHANGES_REQUESTED** [review]

File: test/lib/Streamly/Test/Prelude/Common.hs (hunk 1 above)
Comment (harendra-kumar) [reply test/lib/Streamly/Test/Prelude/Common.hs:1184]: How is it different than `reverse`? They both should be moved?

Comment (pranaysashank) [reply test/lib/Streamly/Test/Prelude/Common.hs:1184]: Yes both of them should be moved to the Ordered ops I think

Comment (harendra-kumar) [reply test/lib/Streamly/Test/Prelude/Common.hs:1184]: I have too much stuff to do, I will just add a comment and merge it for now.

**Pull request merged**